src/bridge/graphs/peg_in.rs

```diff
@@ -58,20 +58,35 @@ impl Display for PegInDepositorStatus {
     }
 }
 
+#[derive(Debug, PartialEq)]
 pub enum PegInVerifierStatus {
-    PegInWait,     // no action required, wait
-    PegInPresign,  // should presign peg-in confirm
-    PegInComplete, // peg-in complete
+    AwaitingDeposit,     // no action required, wait
+    PendingOurNonces,    // the given verifier needs to submit nonces
+    AwaitingNonces,      // the given verifier submitted nonces, awaiting other verifier's nonces
+    PendingOurSignature, // the given verifier needs to submit signature
+    AwaitingSignatures,  // the given verifier submitted signatures, awaiting other verifier's signatures
+    ReadyToSubmit,       // all signatures collected, can now submit
+    Complete,            // peg-in complete
 }
 
 impl Display for PegInVerifierStatus {
     fn fmt(&self, f: &mut Formatter) -> FmtResult {
         match self {
-            PegInVerifierStatus::PegInWait => write!(f, "No action available. Wait..."),
-            PegInVerifierStatus::PegInPresign => {
-                write!(f, "Signature required. Presign peg-in confirm transaction?")
+            PegInVerifierStatus::AwaitingDeposit => write!(f, "Waiting for deposit transaction..."),
```

Comment:
To be more consistent with the rest of the code I suggest:
```suggestion
            PegInVerifierStatus::AwaitingDeposit => write!(f, "Peg-in deposit transaction not confirmed yet. Wait..."),
```